.github/actions/setup/action.yml

```diff
@@ -42,6 +42,35 @@ runs:
         with:
           distribution: 'temurin' # use same JDK distro as in Trino docker images
           java-version: ${{ inputs.java-version }}
+      - name: Get NodeJS version from parent pom
+        shell: bash
+        run: |
+          node_version=$(grep -m 1 '<dep.frontend-node.version>' "pom.xml" | sed -n 's/.*<dep.frontend-node.version>\(.*\)<\/dep.frontend-node.version>.*/\1/p')
+          if [ -n "$node_version" ]; then
+            echo "node version is $node_version"
+            echo "NODE_VERSION=$node_version" >>${GITHUB_ENV}
+            exit 0
+          fi
+      - name: Cache and Restore local NodeJS
+        id: cache-nodejs
+        if: ${{ format('{0}', inputs.cache) == 'true' && env.NODE_VERSION != '' }}
+        uses: actions/cache@v4
```

Comment:
Maybe use actions/setup-node

Reply:
How this is helps? Then I'd need to copy these files and so on

Reply:
Why copy? You change the location of the runtime anyway. This simplifies the workflow.

Reply:
You can't point to the directory outside of the project root.

Reply:
So copy is the only option. That's why the cache is better and more straightforward option